pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -931,6 +931,18 @@ func (tc *TidbCluster) TiCDCLogLevel() string {
 	return "info"
 }
 
-func (tc *TidbCluster) HeterogeneousWithoutLocalPD() bool {
-	return tc.Spec.Cluster != nil && len(tc.Spec.Cluster.Name) > 0 && tc.Spec.PD == nil
+func (tc *TidbCluster) Heterogeneous() bool {
+	return tc.Spec.Cluster != nil && len(tc.Spec.Cluster.Name) > 0
+}
+
+func (tc *TidbCluster) WithoutLocalPD() bool {
+	return tc.Spec.PD == nil
+}
+
+func (tc *TidbCluster) HeterogeneousWithRemote() bool {
+	return tc.Heterogeneous() && tc.Spec.ClusterDomain != ""
```

Comment:
We should check `tc.spec.cluster.clusterDomain`?

Reply:
yes... fix it later